Userland/Applications/SystemMonitor/ProcessModel.cpp

```diff
@@ -469,54 +469,104 @@ void ProcessModel::update()
+                }
+            } else {
+                // NOTE: If there are no threads left in a process this is an indication
+                // for a zombie process, so it should be handled differently.
                 ThreadState state(**process_state);
-                state.tid = thread.tid;
+                state.tid = process.pid;
                 state.pid = process.pid;
                 state.ppid = process.ppid;
                 state.pgid = process.pgid;
                 state.sid = process.sid;
-                state.time_user = thread.time_user;
-                state.time_kernel = thread.time_kernel;
+                state.time_user = 0;
+                state.time_kernel = 0;
                 state.kernel = process.kernel;
                 state.executable = process.executable;
-                state.name = thread.name;
+                state.name = process.name;
                 state.command = read_command_line(process.pid);
                 state.uid = process.uid;
-                state.state = thread.state;
+                state.state = "Zombie";
                 state.user = process.username;
                 state.pledge = process.pledge;
                 state.veil = process.veil;
-                state.cpu = thread.cpu;
-                state.priority = thread.priority;
+                state.cpu = 0;
+                state.priority = 0;
                 state.amount_virtual = process.amount_virtual;
                 state.amount_resident = process.amount_resident;
                 state.amount_dirty_private = process.amount_dirty_private;
                 state.amount_clean_inode = process.amount_clean_inode;
                 state.amount_purgeable_volatile = process.amount_purgeable_volatile;
                 state.amount_purgeable_nonvolatile = process.amount_purgeable_nonvolatile;
-                state.syscall_count = thread.syscall_count;
-                state.inode_faults = thread.inode_faults;
-                state.zero_faults = thread.zero_faults;
-                state.cow_faults = thread.cow_faults;
-                state.unix_socket_read_bytes = thread.unix_socket_read_bytes;
-                state.unix_socket_write_bytes = thread.unix_socket_write_bytes;
-                state.ipv4_socket_read_bytes = thread.ipv4_socket_read_bytes;
-                state.ipv4_socket_write_bytes = thread.ipv4_socket_write_bytes;
-                state.file_read_bytes = thread.file_read_bytes;
-                state.file_write_bytes = thread.file_write_bytes;
+                state.syscall_count = 0;
+                state.inode_faults = 0;
+                state.zero_faults = 0;
+                state.cow_faults = 0;
+                state.unix_socket_read_bytes = 0;
+                state.unix_socket_write_bytes = 0;
+                state.ipv4_socket_read_bytes = 0;
+                state.ipv4_socket_write_bytes = 0;
+                state.file_read_bytes = 0;
+                state.file_write_bytes = 0;
```

Comment:
These are all `ThreadState` default values - maybe not set them again here to reduce the amount of real estate being used / increase readability?